TLM/TLM/Custom/PathFinding/CustomPathManager.cs

```diff
@@ -273,96 +274,110 @@ public class CustomPathManager : PathManager {
+                args.maxLength = 20000f;
             }
 
-            if (ExtPathManager.Instance.FindPathPositionWithSpiralLoop(
-                    pos,
-                    ItemClass.Service.Beautification,
-                    laneTypes,
-                    vehicleTypes,
-                    otherLaneTypes,
-                    otherVehicleTypes,
-                    allowUnderground,
-                    false,
-                    Options.parkingAI
-                        ? GlobalConfig.Instance.ParkingAI.MaxBuildingToPedestrianLaneDistance
-                        : 32f,
-                    out posA,
-                    out _,
-                    out distA,
-                    out _) && distA < minDist) {
-                minDist = distA;
-                position = posA;
+            return CustomCreatePath(out path, ref SimulationManager.instance.m_randomizer, args);
+        }
+
+        /// <summary>
+        /// Converts game VehicleInfo.VehicleType to closest TMPE.API.Traffic.Enums.ExtVehicleType
+        /// </summary>
+        /// <param name="vehicleType"></param>
+        /// <returns></returns>
+        private static ExtVehicleType ConvertToExtVehicleType(VehicleInfo.VehicleType vehicleType) {
+            ExtVehicleType extVehicleType = ExtVehicleType.None;
+            if ((vehicleType & VehicleInfo.VehicleType.Car) != VehicleInfo.VehicleType.None) {
+                extVehicleType = ExtVehicleType.Bus;
             }
 
-            if (allowTransport && ExtPathManager.Instance.FindPathPositionWithSpiralLoop(
-                    pos,
-                    ItemClass.Service.PublicTransport,
-                    laneTypes,
-                    vehicleTypes,
-                    otherLaneTypes,
-                    otherVehicleTypes,
-                    allowUnderground,
-                    false,
-                    Options.parkingAI
-                        ? GlobalConfig
-                          .Instance.ParkingAI.MaxBuildingToPedestrianLaneDistance
-                        : 32f,
-                    out posA,
-                    out _,
-                    out distA,
-                    out _) && distA < minDist) {
-                position = posA;
+            if ((vehicleType & (VehicleInfo.VehicleType.Train | VehicleInfo.VehicleType.Metro |
+                                VehicleInfo.VehicleType.Monorail)) !=
+                VehicleInfo.VehicleType.None) {
+                extVehicleType = ExtVehicleType.PassengerTrain;
             }
 
-            return position.m_segment != 0;
-        }
+            if ((vehicleType & VehicleInfo.VehicleType.Tram) != VehicleInfo.VehicleType.None) {
+                extVehicleType = ExtVehicleType.Tram;
+            }
 
-        /*internal void ResetQueueItem(uint unit) {
-                queueItems[unit].Reset();
-        }*/
+            if ((vehicleType & VehicleInfo.VehicleType.Ship) != VehicleInfo.VehicleType.None) {
```

Comment:
I think vehicleTypeInfo.IsFlagSet(VehicleInfo.VehicleType.Ship) is more readable. probably not worth the effort to change it now. Just a suggestion for the future.